Refactor sidebar renders into methods (prep for mwc-list conversion)

donkawechico opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->

## Proposed change

The sidebar render code is large and can be a bit difficult to reason about. This PR extracts the various "sections" of sidebar into their own render methods for better readability.

This is the first step toward converting sidebar to using mwc-list instead of paper-listbox. Related PR: https://github.com/home-assistant/frontend/pull/7326

<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->

## Type of change

<!--
  What type of change does your PR introduce to the Home Assistant frontend?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [x] Code quality improvements to existing code or addition of tests

## Example configuration

<!--
  Supplying a configuration snippet, makes it easier for a maintainer to test
  your PR.
-->

```yaml

```

## Additional information

<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: https://github.com/home-assistant/frontend/issues/7442
- Link to documentation pull request:

## Checklist

<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

<!--
  Thank you for contributing <3
-->

[docs-repository]: https://github.com/home-assistant/home-assistant.io

donkawechico committed:
Extract renders to methods in preparation for converting paper-listbox to mwc-list

bramkragten commented on src/components/ha-sidebar.ts:
```diff
@@ -202,198 +202,227 @@ class HaSidebar extends LitElement {
     }
 
+    return html`
+      ${this.renderHeader()} ${this.renderItems()} ${this.renderDivider()}
+      ${this.renderNotifications()} ${this.renderUserItem()}
```
```ts
      ${this.renderHeader()}
      ${this.renderPanels()}
      ${this.renderDivider()}
      ${this.renderNotificationItem()}
      ${this.renderUserItem()}
```

donkawechico replied:
It auto-prettifies to this :( :( 

I can work around it by pulling the wrapping <div>s out of those methods and surrounding the call, but that doesn't feel right to me -- a `renderX()` should be fully responsible for rendering its content.

donkawechico replied:
Disabled prettier in two places.

bramkragten commented on src/components/ha-sidebar.ts:
```diff
@@ -202,198 +202,227 @@ class HaSidebar extends LitElement {
+    `;
+  }
+
+  protected renderHeader() {
```
Let's make these private.

donkawechico replied:
Done

bramkragten commented:
You didn't change any actual logic here right?

donkawechico commented:
> You didn't change any actual logic here right?

95% true. I arguably "changed the logic" with how items in edit mode are rendered. That is, I converted this flattened, in-line pattern:

```
protected oneRenderToRuleThemAll() {
  return html`
    <div>Some stuff</div>
    ${this.isEditMode ? "Render items in edit mode" : "Render items in normal mode"}
    <div>Some other stuff</div>
  `
}
```

And refactored into state pattern:

```
protected renderItems() {
  return html`
    this.editMode ? this.renderEditModeItems() : this.renderNormalModeItems()
  `;
}

protected renderEditModeItems() {
  return html`
       <div>Some stuff</div>
      Render items in edit mode
      <div>Some other stuff</div>
  `;
}

protected renderNormalMode() {
  return html`
       <div>Some stuff</div>
      Render items in normal mode
      <div>Some other stuff</div>
  `;
}
```

donkawechico committed:
Make renders private and disable prettier for some lines
